src/main/scala/model/mutation/Mutation.scala

```diff
@@ -0,0 +1,39 @@
+package model.mutation
+
+import model.genome.Genes
+import model.genome.Genes.GeneKind
+
+/**
+ * Model representation of the mutation
+ */
+trait Mutation {
+  val geneKind: GeneKind
+  val isDominant: Boolean
+  // add when introducing factors in the environments
```

Comment:
Questa parte non serve perchè sarà modellata all'interno dei fattori stessi.
Credo che in fase di progettazione del model si fosse deciso di utilizzare il concetto di Mutation solo per rappresentare le mutazioni inserite dall'utente

Reply:
La mutation dovrebbe servire per indicare alle funzioni della `Reproduction` quali geni mutati deve inserire nella successiva generazione, quindi secondo me è corretto avere il trait `Mutation` per passare il GeneKind a cui fa riferimento e la dominanza (da usare per la funzione `setAlleleDominance`)

Reply:
Sì scusate forse mi sono espressa male o ho messo male il commento, sono d'accordo anche io che il trait serva! Io mi riferivo ai metodi che sono stati commentati riguardanti gli effetti dei fattori, quelli secondo me non servono

Reply:
Ops non avevo capito, su questo concordo con te!

Reply:
Lascio il trait ma rimuovo i metodi commentati